routers/web/web.go

```diff
@@ -1584,6 +1585,18 @@ func registerRoutes(m *web.Router) {
 			m.Get("/*", context.RepoRefByType(""), repo.Home) // "/*" route is deprecated, and kept for backward compatibility
 		}, repo.SetEditorconfigIfExists)
 
+		// Add a /tree/* path to redirect to the deprecated /src/*
+		// path. This emulates both Github & Gitlab's URL structure.
```

Comment:
Do they serve responses on `/tree/*`, or do they also redirect?

Reply:
They serve a response. I chose a redirect in order to avoid having duplicate URLs for the same content.

Reply:
Then why should Gitea support it since the link is generated by Web UI? If nobody would write `/tree/` URLs manually, I do not think the newly added routes would bring benefit to real world usages.